lib/pieceprovider/piecepark_reader.go

```diff
@@ -0,0 +1,64 @@
+package pieceprovider
+
+import (
+	"context"
+	"io"
+
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-state-types/abi"
+
+	"github.com/filecoin-project/curio/lib/paths"
+	"github.com/filecoin-project/curio/lib/storiface"
+)
+
+type PieceParkReader struct {
```

Comment:
This is what finds and reads PDP pieces for get requests correct?

Reply:
Yeah, PiecePark is the general "raw piece" storage in Curio, we eventually intend to serve all retrievals from it (instead of from the fr32-padded "unsealed" sector files), but PDP is the first place which will actually do reads directly from it.